src/foam/lib/csv/Outputter.java

```diff
@@ -176,7 +176,7 @@ protected void outputFObject(FObject obj) {
 
     while ( i.hasNext() ) {
       PropertyInfo prop = (PropertyInfo) i.next();
-      prop.toCSV(this, prop.f(obj));
+      output(prop.f(obj));
```

Comment:
Why this change?  shouldn't the property's toCSV still be respected if it wants to control how it outputs itself to CSV?

Reply:
The behaviour of toCSV has changed and isn't really compatible with this Outputter anymore so I'm guessing the safest thing to do was this. If I'm reading this correctly, this roughly preserves the previous behaviour with the exception of Enums. IMO, we should be aiming to nuke this outputter completely as it's replaced by the CSVSink.

Reply:
I see that toCSV now takes a CSVSink?  That doesn't sound right to me.  CSVSink should just internally use a CSVOutputter.  As it stands we're setting outselves up for confusion and possible bugs as people write javaToCSV for properties but then that doesn't actually get called when the object is output the CSV.

Reply:
So short of re-writing this outputter to function like the new outputter, this was the quick fix.

This I feel should be removed. But need to confirm that no one is using it.

The toCSV that this is expecting is not the same as what we have developed here.

And ultimately all this toCSV that was doing was 'output(prop.f(obj))'

Do you think it will suffice to just create a ticket to investigate its use and replace with the new outputter????

Reply:
Yeah the CSVSink is basically an Outputter. Let's do this:
- Rename CSVSink to CSVOutputter
- Rename put to outputFObject (so it's a similar interface to json Outputter)
- Rename reset to flush
- Implement toString which returns csv in js and and sb.toString in java

Then implement a CSVSink which creates an instance of an outputter and on put, calls outputFObject and on eof, does a csv = outputter.toString()

That sound ok?

Reply:
So CSVSink turns to something like this (note that there's no javaCode in here which you'll need to fill in):
```
foam.CLASS({
  package: 'foam.dao',
  name: 'CSVSink',
  extends: 'foam.dao.AbstractSink',

  documentation: 'Sink runs the csv outputter, and contains the resulting string in this.csv',

  properties: [
    {
      class: 'String',
      name: 'csv',
      view: 'foam.u2.tag.TextArea'
    },
    {
      class: 'Class',
      name: 'of',
      visibility: 'HIDDEN'
    },
    {
      name: 'props',
      visibility: 'HIDDEN'
    },
    {
      name: 'outputter',
      transient: true,
      factory: function(props) {
        return this.CSVOutputter.create({
          of: this.of,
          props: this.props
        })
      },
    }
  ],

  methods: [
    function put(obj) {
      this.outputter.outputFObject(obj);
    },

    function eof() {
      this.csv = this.outputter.toString();
    },

    function reset() {
      this.outputter.flush();
      this.csv = '';
    }
  ]
});
```